lib/ui/discover/discover_page.dart

```diff
@@ -32,16 +32,17 @@ class _DiscoveryState extends State<HomeDiscoverPage> {
         return Container(
           child: CircularProgressIndicator(), alignment: Alignment.center);
       case LoadingStatus.SUCCESS:
-        return _buildList(context, viewModel.articlesList, viewModel.goToArticleDetail);
+        return _buildList(context, viewModel.articleDirectory.articles, viewModel.goToDetail);
       case LoadingStatus.ERROR:
         return Text(Strings.errorString);
     }
   }
 }
 
-Widget _buildList(BuildContext context, List<ArticleEntity> articleList, goToArticleDetail) {
+Widget _buildList(BuildContext context, List<ArticleEntity> articlesList, goToArticleDetail) {
   return ListView(
-    padding: const EdgeInsets.only(top: 20.0),
-    children: articleList.map((articleData) => DiscoveryListItem().buildListItem(articleData, goToArticleDetail)).toList(),
-  );
+      padding: const EdgeInsets.only(top: 20.0),
+      children: (articlesList != null) ? (articlesList.map((article) =>
```

Comment:
we could have private function here to make it more readable

Reply:
Do you mean that we can externalise in a private function the mapping or all the "children" itself?